lib/HALTarget/src/Button.h

```diff
@@ -0,0 +1,106 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  Button realization
+ * @author Gabryel Reyes <gabryelrdiaz@gmail.com>
+ *
+ * @addtogroup HALTarget
+ *
+ * @{
+ */
+
+#ifndef BUTTON_H
+#define BUTTON_H
+
+/******************************************************************************
+ * Compile Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "IButton.h"
+#include <stdint.h>
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+
+/** This class provides access to the robot's button. */
+class Button : public IButton
+{
+public:
+    /**
+     * Constructs the button adapter.
+     */
+    Button() : IButton()
+    {
+    }
+
+    /**
+     * Destroys the button adapter.
+     */
+    virtual ~Button()
+    {
+    }
+
+    /**
+     * Is button short pressed or not
+     *
+     * @return If button is short pressed, returns true otherwise false.
+     */
+    bool isShortPressed() final;
+
+    /**
+     * Is button long pressed or not
+     *
+     * @return If button is long pressed, returns true otherwise false.
+     */
+    bool isLongPressed() final;
+
+    /**
+     * Wait until button is released.
+     */
+    void waitForRelease() final;
+
+private:
+    /**
+     * Time to debounce button in milliseconds.
+     */
+    static const uint32_t DEBOUNCE_TIME_MS = 5U;

```

Comment:
Usually too small.

Reply:
20 ms?

Reply:
With 100 ms you are safe. Use at least 40 ms, but don't use blocking debouncing. This destroys your performance.